### PR TITLE
module 1 home task

### PR DESCRIPTION
**Pull request by esteechang**

### CODE REVIEW

File: app.component.ts [hunk 1]
```diff
@@ -1,110 +0,0 @@
...
-    // 1.2. Add API call on each user input. Use mockDataService.getCharacters - to make get request.
-
-    // 2. Since we don't want to spam our service add filter by input value and do not call API until a user enters at least 3 chars.
-
-    // 3. Add debounce to prevent API calls until user stop typing.
-
-    this.charactersResults$ = this.searchTermByCharacters.pipe(
-      // YOUR CODE STARTS HERE
-      map((value: string) => value.trim()),
-      filter((value: string) => value.length === 3),
```
Comment (veerabhadraraochikka) [inline app.component.ts:53]: what if length is grater than 3 ?

File: app.component.ts [hunk 2]
```diff
@@ -1,110 +0,0 @@
...
-  loadCharactersAndPlanet(): void {
-    // 4. On clicking the button 'Load Characters And Planets', it is necessary to process two requests and combine the results of both requests into one result array. As a result, a list with the names of the characters and the names of the planets is displayed on the screen.
-    // Your code should looks like this: this.planetAndCharactersResults$ = /* Your code */
-    // YOUR CODE STARTS HERE
-
-    this.planetAndCharactersResults$ = forkJoin([
-      this.mockDataService.getPlatents(),
-      this.mockDataService.getCharacters(),
-    ]).pipe(
-      map((data) => data.reduce((result, arr) => [...result, ...arr], []))
```
Comment (veerabhadraraochikka) [inline app.component.ts:72]: as we have only two joins then no need to use reduce we can directly use by array extraction

Comment (esteechang) [reply app.component.ts:72]: updated

File: app.component.ts (hunk 1 above)
Comment (esteechang) [reply app.component.ts:53]: updated

**Pull request closed without merging**